From cbe9064e0807031ffb35cf4f1e34a1bf8cb1791c Mon Sep 17 00:00:00 2001
From: Nick Chusid <nchusid@google.com>
Date: Tue, 10 Dec 2024 22:46:28 +0000
Subject: [PATCH] Catch null HuffmanTables when decoding jpeg

Bug: 347735428
Test: TreeHugger
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:a22d1f0f07d998a53dd49a941d3a588f55b36399)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:d474b0f2d958ff94df6af1769188fbf26cabfd92)
Merged-In: Iaf36c076467d18e4e31d8436a6d199c3276a8786
Change-Id: Iaf36c076467d18e4e31d8436a6d199c3276a8786
---
 source/dng_lossless_jpeg.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/dng_lossless_jpeg.cpp b/source/dng_lossless_jpeg.cpp
index 9d0d01a..8802f32 100644
--- a/source/dng_lossless_jpeg.cpp
+++ b/source/dng_lossless_jpeg.cpp
@@ -1616,6 +1616,10 @@ inline int32 dng_lossless_decoder::get_bit ()
 inline int32 dng_lossless_decoder::HuffDecode (HuffmanTable *htbl)
 	{
 	
+	if (htbl == nullptr) {
+		ThrowBadFormat ();
+	}
+
     // If the huffman code is less than 8 bits, we can use the fast
     // table lookup to get its value.  It's more than 8 bits about
     // 3-4% of the time.
-- 
2.40.1

